core/templates/dev/head/pages/exploration_player/AnswerClassificationService.js

```diff
@@ -125,27 +127,19 @@ oppia.factory('AnswerClassificationService', [
           return deferred.promise;
         }
 
-        if (result.outcome === defaultOutcome &&
-            INTERACTION_SPECS[oldState.interaction.id]
-              .is_string_classifier_trainable &&
+        var ruleBasedOutcomeIsDefault = (result.outcome == defaultOutcome);
+        var interactionIsTrainable = INTERACTION_SPECS[
+          oldState.interaction.id].is_string_classifier_trainable;
+        if (ruleBasedOutcomeIsDefault && interactionIsTrainable &&
             ENABLE_ML_CLASSIFIERS) {
-          var classifyUrl = '/explorehandler/classify/' + explorationId;
-          var params = (
-            isInEditorMode ? {} : LearnerParamsService.getAllParams());
-
-          $http.post(classifyUrl, {
-            old_state: oldState.toBackendDict(),
-            params: params,
-            answer: answer
-          }).then(function(response) {
-            var result = response.data;
-            deferred.resolve({
-              outcome: result.outcome,
-              ruleIndex: result.rule_spec_index,
-              answerGroupIndex: result.answer_group_index,
-              classificationCategorization: result.classification_categorization
-            });
-          });
+          var classifierData = StateClassifierMappingService.getClassifierData(
+            stateName);
+          var algorithmId = StateClassifierMappingService.getAlgorithmId(
+            stateName);
+          var predictionService = (
+            PredictionAlgorithmRegistryService.getPredictionService(
+              algorithmId));
+          deferred.resolve(predictionService.predict(classifierData, answer));
```

Comment:
Also do not resolve directly. What if the prediction service's answer is `undefined`? That can happen in code classifier if IE11 is used as browser. So it'd be good to check the prediction before passing it.